libs/core/execution_base/include/hpx/execution_base/completion_signatures.hpp

```diff
@@ -600,6 +606,58 @@ namespace hpx::execution::experimental {
     template <typename Sender, typename Env = empty_env>
     inline constexpr bool is_sender_v = is_sender<Sender, Env>::value;
 
+    namespace detail {
+
+        // Sender may or may not support sends_stopped
+        // In the false case only the value_types and error_types are checked
+        template <class S, class R, class CS,
```

Comment:
Not critical, but more of a consistency issue: in HPX we generally use `typename` instead of `class` for template parameter definitions.

Reply:
Noted, will change it.